oap-server/analyzer/agent-analyzer/src/main/java/org/apache/skywalking/oap/server/analyzer/provider/trace/CacheReadLatencyThresholdsAndWatcher.java

```diff
@@ -0,0 +1,79 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ */
+
+package org.apache.skywalking.oap.server.analyzer.provider.trace;
+
+import com.google.common.base.Splitter;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.concurrent.atomic.AtomicReference;
+import org.apache.skywalking.oap.server.analyzer.module.AnalyzerModule;
+import org.apache.skywalking.oap.server.configuration.api.ConfigChangeWatcher;
+import org.apache.skywalking.oap.server.library.module.ModuleProvider;
+
+public class CacheReadLatencyThresholdsAndWatcher extends ConfigChangeWatcher {
+    private AtomicReference<Map<String, Integer>> thresholds;
+    private final String initialSettingsString;
+    private volatile String dynamicSettingsString;
+
+    public CacheReadLatencyThresholdsAndWatcher(String config, ModuleProvider provider) {
+        super(AnalyzerModule.NAME, provider, "slowCacheReadThreshold");
+        thresholds = new AtomicReference<>(new HashMap<>());
+        initialSettingsString = config;
+
+        activeSetting(config);
+    }
+
+    private void activeSetting(String config) {
```

Comment:
I noticed a potential bug in here, as same as `DBLatencyThresholdsAndWatcher#activeSetting`.

This method doesn't set `default` if it doesn't exist in the provided value. So, in `getThreshold`, a NPE would through as we try to use `thresholds.get().get("default")` for an `int.

We need to fix the existing(also update the change log to add a bug fix), also the new added two watchers.

If there is no default, we should add a default value, as same as we set in the `application.yml`. NOTICE, `initialSettingsString` could be overridden by system env or manual changing. We need to hard code a default value inside this class.

Reply:
Got it . I would fix these bug together .

Reply:
I  have inspect the classes which extends `ConfigChangeWatcher` .  We should update `CacheReadLatencyThresholdsAndWatcher` , `CacheWriteLatencyThresholdsAndWatcher` , `DBLatencyThresholdsAndWatcher`. 
I  think we have two strategy to choose default value  when `get("default")` is Null :

1. Same as `application.yml` 
2. Choose a maximum,  e. g `DBLatencyThresholdsAndWatcher#getThreshold`, we choose `Interger.MAX_VALUE ` as default value  .

Strategy 1 ,  means we use default value from `application.yaml ` if user set default as null .

Strategy 2,  means all condition that depends the  value would be mismatched   (e.g. `if((latency > threshold)` always return false )

Reply:
I would like Strategy 2 .  Because  If user set property null , it's better that we  could skip  some feature(function) ,
and If user want  some feature(function)  , they should specify appropriate property clearly .

Reply:
If  we choose Strategy 2 , developers needn't keep the default value consistent  between hard code  and  `application.yaml`.

Reply:
Make sense.